### ISSUE TITLE
Records: Warn if a get-only property is unused

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#47677]
```csharp
public record A
{
    public string P1 { get; }
}
```

There should be a warning on P1 that it is never set.

See @jnm2's comment (https://github.com/dotnet/roslyn/issues/47666#issuecomment-692054797)

**Comment by HaloFour** [comment]
This situation isn't unique to records:

```cs
public class A
{
    public string P1 { get; }
}
```

The result here is the same: `P1` can only be `null`.

**Comment by Youssef1313** [comment]
@HaloFour, Yes the same does hold for classes. But records may be treated differently in this case for backcompat.

For classes, this is probably a candidate for a warning wave.

**Comment by HaloFour** [comment]
Agreed.  The IDE could at least indicate that the field/property is never assigned and will always have a default value.  I'm kind of surprised that the compiler doesn't warn today for a `readonly` field that is never assigned.

**Comment by Youssef1313** [comment]
>  I'm kind of surprised that the compiler doesn't warn today for a readonly field that is never assigned.

That should be another candidate for a warning wave :smile:

**Comment by jnm2** [comment]
>  I'm kind of surprised that the compiler doesn't warn today for a `readonly` field that is never assigned.

This is already coming in warning waves and it's been shipped for a long time if you put `<Features>strict</Features>` in your Directory.Build.props:

![image](https://user-images.githubusercontent.com/8040367/93103805-114eab00-f67b-11ea-8c30-d03c413df5b6.png)

**Comment by Youssef1313** [comment]
> 
> 
> > I'm kind of surprised that the compiler doesn't warn today for a `readonly` field that is never assigned.
> 
> This is already coming in warning waves. It's been shipped for a long time if you put `<Features>strict</Features>` in your Directory.Build.props:
> 
> ![image](https://user-images.githubusercontent.com/8040367/93103805-114eab00-f67b-11ea-8c30-d03c413df5b6.png)

Aha I wasn't using strict mode.

**Comment by jnm2** [comment]
(IIRC strict mode is undocumented, but the new warning level 5 will include it entirely.)

**Comment by jinujoseph** [comment]
cc @CyrusNajmabadi

**Comment by Youssef1313** [comment]
@jinujoseph I just noticed this is assigned to IDE. However, I meant this to be a compiler warning (this doesn't require a warning wave as records is a new feature). But if the compiler team doesn't think this is a good warning for the compiler, I'm okay with this as an IDE analyzer.

**Comment by jinujoseph** [comment]
Sure @Youssef1313 , looping in @jcouv from Compiler team
